### PR TITLE
raftstore: learner load merge target & fix a merge network recovery bug

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: Liqi Geng <gengliqiii@gmail.com>

###  What have you changed?

1. A learner should load merge target if another overlapped region wants to apply snapshot or create a peer.
2. Remove a condition check in maybe_create_peer. It may lead to a region can't be created forever. Add a test to reproduce this case.

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by youjiali1995** [comment]
@gengliqi  Is it ready for review?

**Comment by gengliqi** [comment]
> @gengliqi Is it ready for review?

Yes.

File: components/raftstore/src/store/fsm/store.rs [hunk 1]
```diff
@@ -2145,6 +2127,27 @@ mod tests {
         assert_eq!(declined_bytes, expected_declined_bytes);
     }
 
+    // check whether the range is covered by existing regions.
+    fn is_range_covered<'a, F: Fn(u64) -> &'a metapb::Region>(
```
Comment (Connor1996) [inline components/raftstore/src/store/fsm/store.rs]: we can remove it and the related test

File: components/raftstore/src/store/peer.rs [hunk 2]
```diff
@@ -2537,6 +2537,32 @@ impl Peer {
             }
         }
     }
+
+    pub fn send_wake_up_message<T: Transport>(&self, trans: &mut T) {
```
Comment (5kbpers) [inline components/raftstore/src/store/peer.rs]: How about `bcast_wake_up_message`

Comment (gengliqi) [reply components/raftstore/src/store/peer.rs]: good name

**Pull request merged**